ipaserver/install/cainstance.py

```diff
@@ -1982,6 +1988,15 @@ def migrate_profiles_to_ldap():
     profile_ids = match.group(1).split(',')
```

Comment:
Commentary related to the profile upgrade / new profile discussion.
Revisiting this code again, we see that:
- Any Dogtag upgrade scripts that *add* new profiles (including referencing them in CS.cfg `profile.list`) will cause FreeIPA to also import that profile to LDAP.  This is fine in general.  If there is a conflict with custom profile name, the custom profile will be left alone.
- Any *updated* Dogtag profiles in the instance profile dir will not be applied in LDAP.

I don't have any comment whether this is good or bad or how it could be improved.  I just wanted to record my understanding of what currently happens in these scenarios.

Reply:
I added more doc strings to the function that explain how migration behaves.

Regarding profile patching / updates: I think updates should be handled by Dogtag. Dogtag knows the previous state of a profile and should know how and when to update a profile.